### ISSUE TITLE
LLDB source location resolution failse since nightly 2022-01-28

### ISSUE THREAD

**Issue by RagibHasin** [rust-lang/rust#93624]
LLDB on Windows is unable to resolve breakpoint to any actual locations since nightly 2022-01-28 build which is working fine with the previous nightly.

### Code

I tried this code:

```rust
fn main() {
    let mut i = 9;
    i -= 3;
    assert_eq!(i, 6);
    println!("{i}");
}
```

I expected to see this happen: 

```
(lldb) target create "D:\\rusty\\target\\debug\\rusty.exe"
Current executable set to 'D:\rusty\target\debug\rusty.exe' (x86_64).
(lldb) b -f main.rs -l 3
Breakpoint 1: where = rusty.exe`static void rusty::main() + 15 at main.rs:3, address = 0x000000014000100f
(lldb)
```

Instead, this happened: 

```
(lldb) target create "D:\\rusty\\target\\debug\\rusty.exe"
Current executable set to 'D:\rusty\target\debug\rusty.exe' (x86_64).
(lldb) b -f main.rs -l 3
Breakpoint 1: no locations (pending).
WARNING:  Unable to resolve breakpoint to any actual locations.
(lldb)
```

### Version it worked on

It most recently worked on: Rust nightly-2022-01-27

`rustc --version --verbose`:
```
rustc 1.60.0-nightly (6abb6385b 2022-01-26)
binary: rustc
commit-hash: 6abb6385b2cb7249f67b9b3ce7522527767dd907
commit-date: 2022-01-26
host: x86_64-pc-windows-msvc
release: 1.60.0-nightly
LLVM version: 13.0.0
```

### Version with regression

It regressed on: Rust nightly-2022-01-28

`rustc --version --verbose`:
```
rustc 1.60.0-nightly (21b4a9cfd 2022-01-27)
binary: rustc
commit-hash: 21b4a9cfdcbb1e76f4b36b5c3cfd64d627285093
commit-date: 2022-01-27
host: x86_64-pc-windows-msvc
release: 1.60.0-nightly
LLVM version: 13.0.0
```

### Backtrace

Not applicable.

Regressed between two nightlies (`2022-01-27` and `2022-01-28`) after Rust 1.58.1

**Comment by ehuss** [comment]
I suspect this is almost certainly from #93081 which includes https://github.com/rust-lang/llvm-project/commit/2abffbf977a9e8c6ca4174a08fe5c4d7781f0aac.  cc @nikic

**Comment by nikic** [comment]
That change actually comes from https://github.com/rust-lang/rust/pull/92767, so cc @arlosi.

**Comment by arlosi** [comment]
@RagibHasin what version of LLDB are you using on Windows. I'm unable to get it to work with stable or nightly (hitting "Unable to resolve breakpoint to any actual locations").

**Comment by RagibHasin** [comment]
@arlosi Regular LLDB bundled with LLVM cannot read Rust debug info, as far as I investigated. I actually faced breakpoint not loading with CodeLLDB first and then upon trying with their bundled LLDB, verified the problem.

**Comment by arlosi** [comment]
So the LLDB included with CodeLLDB has been specially patched to recognize Masm (the previous language identifier) as potentially Rust code. That's what the vanilla LLDB doesn't work before or after, and the CodeLLDB does.

Here's the commit that makes it work in CodeLLDB's LLVM fork: https://github.com/vadimcn/llvm-project/commit/4aba5a4d27585ae8d4c3c94dc33da3aa055bc664

I've submitted a patch to LLDB to get it working in the latest LLDB: https://reviews.llvm.org/D119044

I created an issue on the CodeLLDB GitHub as well https://github.com/vadimcn/vscode-lldb/issues/633 (cc @vadimcn) to get a fix in the patched version over there.

Since VSCode extensions can be quickly updated, I assuming we can get a patched version out soon, I don't think we need to revert the change.

**Comment by RagibHasin** [comment]
Thank you very much @arlosi . I actually wasn't sure if it is an issue of Rust itself or LLDB/CodeLLDB. So, decided to raise it here. Hope it resolves soon.

**Comment by Jeremiah-Griffin** [comment]
I seem to be experiencing this now in the 1.59 stable release and not the 1.60 nightly.

Same issues were detailed here:
https://github.com/vadimcn/vscode-lldb/issues/641#issue-1130568438

and here:
https://github.com/vadimcn/vscode-lldb/issues/633#issue-1124725975

Bit new to GitHub so if you need more detail I would be happy to supply it.

**Comment by F0903** [comment]
> I seem to be experiencing this now in the 1.59 stable release and not the 1.60 nightly.
> 
> The issues seems to express just as is detailed here: [vadimcn/vscode-lldb#641 (comment)](https://github.com/vadimcn/vscode-lldb/issues/641#issue-1130568438)
> 
> and here: [vadimcn/vscode-lldb#633 (comment)](https://github.com/vadimcn/vscode-lldb/issues/633#issue-1124725975)
> 
> A rebuild on a version as late as 1.58.1 stable fixes the issue.
> 
> Bit new to GitHub so if you need more detail I would be happy to supply it.

I also have this issue on 1.59.

**Comment by arlosi** [comment]
This ended up getting (unintentionally) backported to the 1.59.0 stable release along with another unrelated LLVM change. More details are available over on the CodeLLDB page here: https://github.com/vadimcn/vscode-lldb/issues/633#issuecomment-1057209602

We need to get a new release of the CodeLLDB extension to fix it. Until a new CodeLLDB is available the options are:
* Use the 1.58.1 toolchain.
* Use the `-gnu` toolchain.
* Compile CodeLLDB and LLDB (which has the fix now) yourself.

**Comment by Jeremiah-Griffin** [comment]
> This ended up getting (unintentionally) backported to the 1.59.0 stable release along with another unrelated LLVM change. More details are available over on the CodeLLDB page here: https://github.com/vadimcn/vscode-lldb/issues/633#issuecomment-1057209602
> 
> We need to get a new release of the CodeLLDB extension to fix it. Until a new CodeLLDB is available the options are:
> * Use the 1.58.1 toolchain.
> * Use the `-gnu` toolchain.
> * Compile CodeLLDB and LLDB (which has the fix now) yourself.

Thank you kindly for taking the time to note this - have a good one!

**Comment by RagibHasin** [comment]
@arlosi is there any instruction about how to compile Rust-enabled LLDB somewhere? It seems that I cannot find it. Some pointers would be very helpful, for me and for future reference also, I think.

**Comment by johnw42** [comment]
I think this is worth making a new 1.59.x release to fix. Discovering what's going wrong is not easy. I was trying to debug Rust code on a new computer, and I assumed I was doing something wrong. I was starting to doubt my sanity until I tried debugging a Rust program on an older computer (where I had 1.57 installed) and it worked.

**Comment by arlosi** [comment]
There's a prerelease version of CodeLLDB that has a fix for this: https://github.com/vadimcn/vscode-lldb/issues/633#issuecomment-1059681468

You can download the appropriate .vsix file, and install it in VSCode:
<img width="466" alt="image" src="https://user-images.githubusercontent.com/704597/157083359-dc01211c-2504-4492-9a95-42a3cbb89f1c.png">

Please report any feedback / issues over on the CodeLLDB thread. Hopefully we have a new official release soon.

**Comment by arlosi** [comment]
CodeLLDB has released an updated version (1.7.0) that fixes the issue. This can be closed.

**Comment by RagibHasin** [comment]
Yes. Thanks, everyone, for keeping Rust running smoothly.
